### PR TITLE
Support multiple path arguments on the command line

### LINKED ISSUE #11: Allow multiple packages to be passed to program
It would be helpful if multiple packages could be provided to `ineffassign` for analysis. Most other static Go analysis tools support this behavior as well, so would be great to make this consistent. Happy to open a PR if you think this would be reasonable.

### PR DESCRIPTION
**Pull request by dnephin**
Fixes #11 

Also exits with non-zero if there is an error with the arguments.

### CODE REVIEW

**Comment by dnephin** [comment]
This fixes a problem we have when integrating with gometalinter (https://github.com/alecthomas/gometalinter/issues/342)

File: ineffassign.go [hunk 1]
```diff
@@ -12,26 +12,32 @@ import (
...
+var dontRecurseFlag = flag.Bool("n", false, "don't recursively check paths")
 
 func main() {
 	flag.Parse()
-	if len(flag.Args()) != 1 {
+
+	if len(flag.Args()) == 0 {
 		fmt.Println("missing argument: filepath")
-		return
+		os.Exit(3)
```
Comment (gordonklaus) [inline ineffassign.go]: What is the meaning of exit code 3?  Perhaps this should be in a named constant.

File: ineffassign.go [hunk 2]
```diff
@@ -12,26 +12,32 @@ import (
...
-	root, err = filepath.Abs(flag.Arg(0))
-	if err != nil {
-		fmt.Printf("Error finding absolute path :%s", err)
-		return
+	lintFailed := false
+	for _, path := range flag.Args() {
+		root, err := filepath.Abs(path)
+		if err != nil {
+			fmt.Printf("Error finding absolute path: %s", err)
+			os.Exit(3)
```
Comment (gordonklaus) [inline ineffassign.go]: same as above

File: ineffassign.go [hunk 3]
```diff
@@ -12,26 +12,32 @@ import (
...
+		}
+		lintFailed = walkPath(root) || lintFailed
```
Comment (gordonklaus) [inline ineffassign.go]: An explicit `if walkPath(root) { lintFailed = true }` would be more readable than short-circuit `||`.

**Comment by dnephin** [comment]
Cool, I made those changes

**Comment by dnephin** [comment]
Thanks!

**Comment by gordonklaus** [comment]
No prob 😄

**Pull request merged**